### ISSUE TITLE
route goes via fence

### ISSUE THREAD

**Issue by Tsurha** [IsraelHikingMap/Site#1571]
## What I expect to happen
That the route does not go via a fence, but goes around

## What really happened
It travels through, see pic below

## What I did that caused the issue - step by step
1. Added a new fence to the map where it exists and checked that it appears on the map. It did instantly.
2. Waited one day, giving a chance to the routing engine to read the new fence data
3. Tried routing on the old trail that was now blocked by a fence. Unfortunately it did.

## My environment
App on Android

## Things I think the developers should know, images, links etc.
![Screenshot_20210816-083630](https://user-images.githubusercontent.com/49042192/129516303-781dfcc8-1e05-4dc4-9b0e-f64cdaa8d378.jpg)

**Comment by HarelM** [comment]
Please include a location. I would check the mapping. If there's a path there it seems like a gate is missing...

**Comment by Tsurha** [comment]
Apologies for failing to add the link, now below. The path existed on the ground and the the fence is new, without a gate there.  They just built a fence on the path which still exists on both its sides.  I added the fence as it is in reality and expected the route to go around it.

https://israelhiking.osm.org.il/map/18.66/32.2211/34.8731

**Comment by zstadler** [comment]
@Tsurha,
 
Please a `barrier` node, optionally with the appropriate `access` tag, when mapping linear barrier that cross `highway` ways.

If there is no node in the intersection of the `highway` way and a linear `barrier`, or if their intersection node does not have a `barrier` tag, then the routing will not be interrupted by the linear barrier.

**Comment by Tsurha** [comment]
Added a barrier=yes. It still routes through it.

![image](https://user-images.githubusercontent.com/49042192/129911674-0bf8b536-70ab-4914-a722-0eff07b9941f.png)

**Comment by HarelM** [comment]
You need to add `access=no`...

**Comment by Tsurha** [comment]
Added access=no and the barrier color has changed to red. However the routing algorithm does not care:

![Screenshot_20210819-082412](https://user-images.githubusercontent.com/49042192/130012976-b5f520b1-d5ed-43bb-95f0-8247cbbdd126.jpg)




While to color of the gate was changed, something else did not.  Could my routing thing be related to things that do not update from OSM?  Yesterday I fixed the street name here to Hamanim המנים, deleting one small Yod letter. 
https://www.openstreetmap.org/edit#map=19/32.21857/34.86810
However in IHM its still Haminim המינים

![Screenshot_20210819-081003](https://user-images.githubusercontent.com/49042192/130012925-7a38e88d-2fe8-4616-9750-bb1594b402a5.jpg)

**Comment by HarelM** [comment]
We'll check about the routing. The street name in OSM is still המינים:
![image](https://user-images.githubusercontent.com/3269297/130017808-259799ba-f8de-456e-adf2-6439b5e40c28.png)

**Comment by HarelM** [comment]
The gate node is not part of the highway in OSM (it is only part of the fence and looks like it is on to of the highway but not part of it) - to map a barrier correctly you need the node to be part of the highway and part of the fence.
Routing is updated hourly, so you can check an hour after you update...

**Comment by zstadler** [comment]
@Tsurha,
The routing algorithms only consider the `highway` lines, their tags, and the tags of the nodes belonging to these lines. When adding a barrier, please make sure it is created on the highway.

Also, is there a gate, which can be opened by someone with the key, or does the new fence simply blocks the path? The current tagging describes the first option. For the second options, a `barrier=yes` tag is more appropriate.

**Comment by Tsurha** [comment]
Bottom line - not solved.

@HarelM - You were spot on in recognizing that the barrier was not exactly on the "highway". I had to zoom up to 24 and moved it a few hours ago. Its exactly on the bot the path and the fence now.  

@zstadler - The new fence simply blocks the path so per your suggestion its  a barrier=yes tag. When adding it, by selecting Barrier at the top, the editor does not suggest the access tag. I had to add manually it in the bottom with access=no.  Would that be the reason the route still goes through?

The current tag is in the picture for your review :
![image](https://user-images.githubusercontent.com/49042192/130128317-fc5bea14-ade4-4201-b77a-1fcbf52ee596.png)

**Comment by zstadler** [comment]
Apparently, the routing ignores `barrier=yes` nodes.
https://www.openstreetmap.org/node/9016049986

On second thought, if there is no way of going through the fence, the path should probably be disconnected.

**Comment by Tsurha** [comment]
Indeed I can fix this place by deleting a part of the path. Aside from locally fixing there, its a bug for your code fixing consideration. Farmers constantly add barriers in Israel, regardless of their right to do so as they do face theft. If a barrier is mapped, the auto route should not take you through it. Users could use a straight line if they'd knowingly decide to.

**Comment by zstadler** [comment]
Although `barrier=fence` on a node is [not recommended according to the wiki](https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dfence), it is [supported by the Graphhopper routing engine](https://github.com/graphhopper/graphhopper/issues/1468).

BTW, if the barrier is a gate that the farmer can open, then a `barrier=gate` with `access=no` is appropriate.

**Comment by Tsurha** [comment]
I've deleted the little path connecting the road to the fence.  This specific Beni Zion point is solved as far as mapping and routing. 
I'll seperatly mourn the uphill path it led to.  This hill gave us wonderful interval trainings and is now unusual.  I'll send regards in behalf of IHM to the rich guy who is building a mansion there...

Regarding Graphhopper ignoring general barrier=yes nodes, to me its a bug.  I'd rather not open it myself there as my usage is limited to IHM, I have no clue how you use Graphhopper, its forks or versions.  @zstadler would you open it?

**Comment by zstadler** [comment]
@Tsurha, it is up to you to suggest and justify the change you desire in Graphhopper.
